src/mono/mono/metadata/marshal.c

```diff
@@ -6264,6 +6264,31 @@ mono_marshal_asany_impl (MonoObjectHandle o, MonoMarshalNative string_encoding,
 
 		return res;
 	}
+	case MONO_TYPE_SZARRAY: {
+		MonoClass *klass = t->data.klass;
+		MonoClass *eklass = m_class_get_element_class (klass);
+		MonoArray *arr = (MonoArray *) MONO_HANDLE_RAW (o);
+
+		if (m_class_get_rank (klass) > 1)
+			break;
+
+		if (arr->bounds)
```

Comment:
Shouldn't this always be NULL for SZArrays?

Reply:
Not sure, I thought bounds and sized arrays are two distinct array features and a SZArray could have bounds.